proto/tendermint/abci/types.proto

```diff
@@ -190,14 +228,15 @@ message ResponseCheckTx {
   bytes          data       = 2;
   string         log        = 3;  // nondeterministic
   string         info       = 4;  // nondeterministic
-  int64          gas_wanted = 5 [json_name = "gas_wanted"];
-  int64          gas_used   = 6 [json_name = "gas_used"];
+  int64          gas_wanted = 5;
```

Comment:
Is gas configuration likely to grow/change in the future? Might we want a message type here (e.g., `GasOptions` or something)? 

I mention this because I see we duplicate these fields in a couple of places, and when there are multiple related bits that need to move together I think "message".

Reply:
AFAICT, the discussions I've been part of in the context of ABCI++ the idea was to keep the concept of *gas* unmodified, albeit, providing parameters in the new APIs if that makes sense.

I would personally keep it as it is now because:

a) as I said above *gas* is not likely to change
b) as `ResponseCheckTx` defined those parameters like this long ago, App developers are used to this

Reply:
I'm torn here because I really like the idea of an encapsulated, easily extensible `gas_options` field. It's not related to ABCI++ though, so perhaps we should consider breaking this out into a separate issue that we can address at another time (perhaps when someone asks for an additional gas-related field?).

Reply:
Thanks @thanethomson  for looking into this. May I suggest you to create an issue with this dilemma? If you do create it, you might consider to include it in the ABCI++ board (toward the end) and then when ABCI++ implementation is move advanced we can pick it up and see when/how we address this

Reply:
My personal view on this is that `gas` is one of the unique cases were the abstraction between application and consensus engine became blurred. "Gas" is an application defined concept which tries to encapsulate the collective cost to the machines running that network to execute and store a transaction. It's part of a systems economics not part of state machine replication. I think this was initially added because people submitting a tx have to go through Tendermint to the application and back, however if a tx doesn't have enough "gas" this doesn't need to be it's own fields rather it should use `codespace`, `data` and `info`. As we are modifying the methods I think it's as good a time as any to take a closer look at the structs these methods return

Reply:
I'd propose to open an issue whose goal is deciding what to do here, and implement the decision in a PR. Then, we can add the issue to the board, and prioritize it.